site/app/controllers/calendar/CalendarController.php

```diff
@@ -78,52 +51,75 @@ public function createMessage(): RedirectResponse {
         }
         else {
             $this->core->addErrorMessage("Invalid or incorrect type given");
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
 
         if (isset($_POST['date'])) {
             $date = $_POST['date'];
         }
         else {
             $this->core->addErrorMessage("Invalid or incorrect date given");
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
 
         if (isset($_POST['text'])) {
             $text = $_POST['text'];
         }
         else {
             $this->core->addErrorMessage("Invalid or incorrect text given");
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
 
         $calendar_item = new CalendarItem();
         try {
             $calendar_item->setStringType($type);
         }
         catch (\InvalidArgumentException $e) {
-            $this->core->addErrorMessage($e->getMessage());
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+            $this->core->addErrorMessage("That is not a valid calendar item type");
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
         $calendar_item->setDate(new \DateTime($date));
         try {
             $calendar_item->setText($text);
         }
         catch (\InvalidArgumentException $e) {
-            $this->core->addErrorMessage($e->getMessage());
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+            $this->core->addErrorMessage("Text exceeds 255 characters which is not allowed");
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
+        }
+
+        if (isset($_POST['course'])) {
+            $set_course = $_POST['course'];
+        }
+        else {
```

Comment:
Is it possible to instead of redirecting them back to the calendar page, to instead just show an error message and let them continue? I see this happen a lot and it could probably get annoying. Again if its too much work a future PR could try to get around this.

Reply:
This is definitely a good suggestion. I think that for now, this is a good start. If this is a problem that wants to be addressed in the future, it could become its own PR.